clang/lib/Sema/TreeTransform.h

```diff
@@ -7668,6 +7668,39 @@ QualType TreeTransform<Derived>::TransformAttributedType(TypeLocBuilder &TLB,
       }
     }
 
+    /* TO_UPSTREAM(BoundsSafety) ON*/
+    // This is primarily here for the `TemplateInstantiator` subclass
+    // where we want to check that after template instantiation the
+    // bounds-safety attributes are being applied to pointer types.
+    if (SemaRef.LangOpts.isBoundsSafetyAttributeOnlyMode()) {
+      const char *AttrName = nullptr;
+      switch (TL.getAttrKind()) {
+      case attr::PtrSingle:
+        AttrName = "__single";
+        LLVM_FALLTHROUGH;
+      case attr::PtrUnsafeIndexable: {
+        AttrName = AttrName ? AttrName : "__unsafe_indexable";
+        QualType ShouldBePtrTy =
```

Comment:
Instead of having a fallthrough, could this live outside of the switch statement, guarded by a check whether `AttrName` is nonnull?

Reply:
I decide to change it what you suggested.